features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/BulkingIT.java

```diff
@@ -0,0 +1,235 @@
+
+public class BulkingIT {
+    private static Logger LOG = LoggerFactory.getLogger(BulkingIT.class);
+
+    @Rule
+    public ElasticSearchRule elasticSearchRule = new ElasticSearchRule(new ElasticSearchServerConfig()
+            .withStartDelay(0)
+            .withManualStartup()
+    );
+
+    private List<Flow> createMockedFlows(final int count) {
+        final List<Flow> flows = new ArrayList<>(count);
+        for (int i = 0; i < count; i++) {
+            final Flow flow = mock(Flow.class);
+            when(flow.getNetflowVersion()).thenReturn(Flow.NetflowVersion.V5);
+            when(flow.getIpProtocolVersion()).thenReturn(4);
+            when(flow.getSrcAddr()).thenReturn("192.168." + i / 256 + "." + i % 256);
+            when(flow.getDstAddr()).thenReturn("192.168." + i / 256 + "." + i % 256);
+            when(flow.getSrcAddrHostname()).thenReturn(Optional.empty());
+            when(flow.getDstAddrHostname()).thenReturn(Optional.empty());
+            when(flow.getNextHopHostname()).thenReturn(Optional.empty());
+            when(flow.getVlan()).thenReturn(null);
+            flows.add(flow);
+        }
+        return flows;
+    }
+
+    private FlowRepository createFlowRepository(final JestClient jestClient, final DocumentEnricher documentEnricher, int bulkSize, int bulkFlushMs) {
+        final ElasticFlowRepository elasticFlowRepository = new ElasticFlowRepository(new MetricRegistry(), jestClient,
+                IndexStrategy.MONTHLY, documentEnricher, new MockSessionUtils(), new MockNodeDao(), new MockSnmpInterfaceDao(),
+                new MockIdentity(), new MockTracerRegistry(), new MockDocumentForwarder(), new IndexSettings());
+        elasticFlowRepository.setBulkSize(bulkSize);
+        elasticFlowRepository.setBulkFlushMs(bulkFlushMs);
+        return new InitializingFlowRepository(elasticFlowRepository, jestClient);
+    }
+
+    /**
+     * Tests that small bulk will be persisted after the given timeout of 1000ms (bulkFlushMs).
+     */
+    @Test
+    public void testFlushTimeout() throws Exception {
+        elasticSearchRule.startServer();
+
+        final RestClientFactory restClientFactory = new RestClientFactory(elasticSearchRule.getUrl());
+
+        try (final JestClient jestClient = restClientFactory.createClient()) {
+            final MockDocumentEnricherFactory mockDocumentEnricherFactory = new MockDocumentEnricherFactory();
+            final DocumentEnricher documentEnricher = mockDocumentEnricherFactory.getEnricher();
+            final FlowRepository flowRepository = createFlowRepository(jestClient, documentEnricher, 1000, 1000);
+
+            final long[] persists = new long[2];
+
+            // send full bulk in order to estimate last persist
+            flowRepository.persist(Lists.newArrayList(createMockedFlows(1000)), FlowDocumentTest.getMockFlowSource());
+
+            with().pollInterval(25, MILLISECONDS).await().atMost(10, SECONDS).until(() -> {
```

Comment:
nitpick on this test :  25 milliseconds is too small to check if it already processed. First iteration will always succeed here. May be we can increase `bulkflushMs` to 5 secs and wait for at least 2 seconds with `pollDelay(2, SECONDS)`.

For next check, we can use `.pollDelay(5, SECONDS)` to create sufficient delay.

Reply:
done.